source/h1_connection.c

```diff
@@ -290,6 +293,12 @@ static bool s_connection_is_open(const struct aws_http_connection *connection_ba
     return is_open;
 }
 
+static bool s_connection_new_requests_allowed(const struct aws_http_connection *connection_base) {
+    struct h1_connection *connection = AWS_CONTAINER_OF(connection_base, struct h1_connection, base);
+    int new_stream_error_code = (int)aws_atomic_load_int(&connection->atomic.new_stream_error_code);
+    return new_stream_error_code == 0;
```

Comment:
If you're feeling daring, you could set the new_stream_error_code to AWS_ERROR_HTTP_CONNECTION_CLOSED anywhere you see `stream->is_final_stream = true;`

that happens whenever we encounter a "connection: close" header